### ISSUE TITLE
OpenBSD (amd64 and arm64) - sys/statfs.rs - fix 

### ISSUE THREAD

**Issue by LouisJasbetz** [nix-rust/nix#1186]
Hi,

File:
`sys/statfs.rs`

```
target_os = "openbsd"
-----------------------------------
pub fn optimal_transfer_size(&self) -> i32 {
    |         self.0.f_iosize
    |         ^^^^^^^^^^^^^^^ expected i32, found u32
help: you can convert an `u32` to `i32` and panic if the converted value wouldn't fit
-----------------------------------
```

```
target_os = "openbsd"
-----------------------------------
pub fn files_free(&self) -> i64 {
    |         self.0.f_ffree
    |         ^^^^^^^^^^^^^^ expected i64, found u64
help: you can convert an `u64` to `i64` and panic if the converted value wouldn't fit
-----------------------------------
```

please fix to:
```
self.0.f_iosize as i32
self.0.f_ffree as i64
```

Regards,
Louis Jasbetz

**Comment by asomers** [comment]
Can you please clarify which version of OpenBSD you're using?  And if you could submit a PR too, that would be great.

**Comment by Artoria2e5** [comment]
Commit https://github.com/nix-rust/nix/commit/7a1e86131361b5fae03517513d46c188399fa07b should have fixed this issue since v0.17.0. Update your crate.

According to the online blame at https://github.com/openbsd/src/blame/f0801dd35b987df069e806e4afc6f87aa5da4038/sys/sys/mount.h, f_iosize has been u32 and f_free has been u64 for 12 years.

**Comment by asomers** [comment]
Closing as duplicate of #1125 .
